pysph/base/particle_array.pyx

```diff
@@ -569,6 +569,13 @@ cdef class ParticleArray:
         prop = ''
         for prop in particle_props:
             self._check_property(prop)
+            if prop in self.stride.keys():
+                stride = self.stride[prop]
+                prop_len = len(particle_props[prop])
+                base_len = len(particle_props['x'])
```

Comment:
You can put this out of the loop and find the number of real particles as self.get_number_of_particles(), do you need the ghost particles also?

Reply:
I have put this inside loop as only the `prop` in `particle_props` need to be checked for the consistency. Poperties which are not present in `particle_props` will be initialized consistently with the stride value later in the function. No, I don't need ghost particles.

`self.get_number_of_particles()` will give me length of the current particle array. However, I want the length of `x` in the new particle_array which is passed.

Reply:
You also do not need to do `if prop in self.stride.keys`, just `prop in self.stride` will do.

Reply:
I mean you can get the base_len outside the loop.  Also what if the particle_props do not have 'x'?  What then?

Reply:
the new particles to be added should at least contain x, y and z. isn't it?
if not we should add that check too.